source/posix/thread.c

```diff
@@ -460,3 +460,16 @@ int aws_thread_current_at_exit(aws_thread_atexit_fn *callback, void *user_data)
     tl_wrapper->atexit = cb;
     return AWS_OP_SUCCESS;
 }
+
+struct aws_string *aws_thread_current_name(struct aws_allocator *allocator) {
+#ifdef AWS_PTHREAD_HAS_GETNAME
+    char name[16] = {0};
+    if (pthread_getname_np(aws_thread_current_thread_id(), name, 16)) {
+        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+        return NULL;
+    }
+    return aws_string_new_from_c_str(allocator, name);
+#else
+    return NULL;
```

Comment:
aws_raise_error(AWS_ERROR_PLATFORM_NOT_SUPPORTED) is the "Feature not supported on this platform" error